central/node/datastore/store/postgres/store.go

```diff
@@ -102,29 +102,48 @@ func (s *storeImpl) insertIntoNodes(
 		return marshalErr
 	}
 
-	values := []interface{}{
-		// parent primary keys start
-		pgutils.NilOrUUID(cloned.GetId()),
-		cloned.GetName(),
-		pgutils.NilOrUUID(cloned.GetClusterId()),
-		cloned.GetClusterName(),
-		cloned.GetLabels(),
-		cloned.GetAnnotations(),
-		pgutils.NilOrTime(cloned.GetJoinedAt()),
-		cloned.GetContainerRuntime().GetVersion(),
-		cloned.GetOsImage(),
-		pgutils.NilOrTime(cloned.GetLastUpdated()),
-		pgutils.NilOrTime(cloned.GetScan().GetScanTime()),
-		cloned.GetComponents(),
-		cloned.GetCves(),
-		cloned.GetFixableCves(),
-		cloned.GetRiskScore(),
-		cloned.GetTopCvss(),
-		serialized,
-	}
-
-	finalStr := "INSERT INTO nodes (Id, Name, ClusterId, ClusterName, Labels, Annotations, JoinedAt, ContainerRuntime_Version, OsImage, LastUpdated, Scan_ScanTime, Components, Cves, FixableCves, RiskScore, TopCvss, serialized) VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16, $17) ON CONFLICT(Id) DO UPDATE SET Id = EXCLUDED.Id, Name = EXCLUDED.Name, ClusterId = EXCLUDED.ClusterId, ClusterName = EXCLUDED.ClusterName, Labels = EXCLUDED.Labels, Annotations = EXCLUDED.Annotations, JoinedAt = EXCLUDED.JoinedAt, ContainerRuntime_Version = EXCLUDED.ContainerRuntime_Version, OsImage = EXCLUDED.OsImage, LastUpdated = EXCLUDED.LastUpdated, Scan_ScanTime = EXCLUDED.Scan_ScanTime, Components = EXCLUDED.Components, Cves = EXCLUDED.Cves, FixableCves = EXCLUDED.FixableCves, RiskScore = EXCLUDED.RiskScore, TopCvss = EXCLUDED.TopCvss, serialized = EXCLUDED.serialized"
-	_, err := tx.Exec(ctx, finalStr, values...)
+	values := map[bool][]interface{}{
+		false: {
+			// parent primary keys start
+			pgutils.NilOrUUID(cloned.GetId()),
+			cloned.GetName(),
+			pgutils.NilOrUUID(cloned.GetClusterId()),
+			cloned.GetClusterName(),
+			cloned.GetLabels(),
+			cloned.GetAnnotations(),
+			pgutils.NilOrTime(cloned.GetJoinedAt()),
+			cloned.GetContainerRuntime().GetVersion(),
+			cloned.GetOsImage(),
+			pgutils.NilOrTime(cloned.GetLastUpdated()),
+			serialized, // TODO: when scan is ignored, we should update it only selectively!
```

Comment:
Any ideas on how we could update `serialized` selectively?

Reply:
Leverage the existing mechanism maybe?

- Fetch node obj from DB in `isUpdated`
- Clone the node object that is provided as the argument to `upsert`
- Pick what to keep from the clone and what to take from the the arg object
- Serialize the clone
- Upsert